### DISCUSSION TITLE
Lesson 4: Problem calling getPriceFeed()

### DISCUSSION THREAD

**Discussion by phvaz** [Cyfrin/foundry-full-course-cu#935] [Q&A] [answered]
So, in Lesson 4(https://sepolia.etherscan.io/address/0x1b30DA2a868704483143A4D46865Ac9585629fd0#readContract) I believe the correct path to the solution is to call getPriceFeeds so the function can return the price of Ethereum, but I have no ideia how to correctly call it. In Remix the function returns its own address istead of the current price and I have no ideia how to proceed. Idk maybe my line of thought is wrong from the beginning but i dont think that's the case based in the answers i've seen here in other questions. 
![Captura de tela 2023-09-28 094211](https://github.com/Cyfrin/foundry-full-course-f23/assets/116604133/5035248e-1324-4c50-a51e-f0a3ea2a34d0)

**Comment by theirrationalone** [comment]
You're returning an address from your `getPricefeed` function. Which is in-fact the correct semantically.

What we expect ?? What should be `pricefeed` to be?? It should be an address pointing to the contract onto the live chain. Right! 

If we wanna `latestPrice` == `latestAnswer` from `priceFeed` we need to create an associated function to get that. 

For example: Somewhere into PriceConverter.sol file. 


```solidity
function getLatestPrice(address pricefeedAddress) internal view returns(uint256) {
    AggregatorV3Interface pricefeed = AggregatorV3Interface(pricefeedAddress);
    (, int256 latestAnswer,,,) = pricefeed.latestRoundData();
    return uint256(latestAnswer);
}
```

After creating this function, You can call it into your `FundMe.sol` appropriately. 

;`)

**Reply by phvaz** [reply]
Thx mate, that was my initial doubt: if i had to increment the code or not(since all the other lessons I could solve using sepolia only). Now I receive the correct result(166940640000 as of now) but when i submit the answer in priceGuess my contract returns the following error: Fail with Custom Error 'LessonFour__WrongPrice ()'

**Reply by theirrationalone** [reply, marked as answer]
Actually it's a price without any precision: 166940640000. As we know it's $1669 and all other afterward are decimal  . Values. 

Since we're working with solidity, we need some precision to handle underflow/overflow on division/multiplication. So here we should make it in form of something like ether value. In fact 1 ether is == 1e18, but we have what?? After 1669 we have 1e8. The output might be this: 1669e8.

Therefore, We need to add 1e10 as a precision to make it always divisible Or multiplied by 1e18 to ensure it will always get divisible by 1e18 without occurring underflow/overflow.

Keep in mind always: we're doing this because it's a way to not run into underflow/overflow. We shall always calculating values in terms of ether. 🍀

**Reply by phvaz** [reply]
Thanks for the help, managed to complete it
